src/core/server/saved_objects/export/collect_exported_objects.ts

```diff
@@ -126,3 +165,68 @@ const fetchReferences = async ({
       .map((obj) => ({ type: obj.type, id: obj.id })),
   };
 };
+
+const buildTransforms = (typeRegistry: ISavedObjectTypeRegistry) =>
+  typeRegistry.getAllTypes().reduce((transforms, type) => {
+    if (type.management?.onExport) {
+      return {
+        ...transforms,
+        [type.name]: type.management.onExport,
+      };
+    }
+    return transforms;
+  }, {} as Record<string, SavedObjectsExportTransform>);
+
+const buildIsExportable = (
+  typeRegistry: ISavedObjectTypeRegistry
+): SavedObjectsExportablePredicate<any> => {
+  const exportablePerType = typeRegistry.getAllTypes().reduce((transforms, type) => {
+    if (type.management?.isExportable) {
+      transforms[type.name] = type.management.isExportable;
+    }
+    return transforms;
+  }, {} as Record<string, SavedObjectsExportablePredicate>);
+
+  return (obj: SavedObject) => {
+    const typePredicate = exportablePerType[obj.type];
+    return typePredicate ? typePredicate(obj) : true;
+  };
+};
+
+const splitByExportability = async (
+  objects: SavedObject[],
+  isExportable: SavedObjectsExportablePredicate<any>
+) => {
+  const exportableObjects: SavedObject[] = [];
+  const nonExportableObjects: SavedObject[] = [];
+  await Promise.all(
+    objects.map(async (obj) => {
+      const exportable = await isExportable(obj);
```

Comment:
If saved object migrations have taught me anything, it's that plugin authors can/will create buggy functions to interrogate their saved objects. Can we be defensive here, and handle any thrown errors explicitly? I'm open to suggestions, but I'm thinking we could either:
1) fail open, and treat failed checks as "exportable"
2) fail closed, and treat failed checks as "unexportable"
3) raise our own error in response to a failed check

Option 3 is my least favorite, because it prevents administrators from exporting saved objects due to a bug in our code, without any recourse. That said, I don't know the intent of preventing objects from being excluded, so I don't know if it's safe for us to always fail open/closed.

Perhaps the safest option is to "fail closed", and report the failure within the export metadata and/or in a server log entry. What do you think?

Reply:
> If saved object migrations have taught me anything, it's that plugin authors can/will create buggy functions to interrogate their saved objects

Can't disagree on that

> Perhaps the safest option is to "fail closed", and report the failure within the export metadata and/or in a server log entry

That's a perfect use case for the currently unused `SavedObjectsExportExcludedObject.reason`